internal/aws/cwlogs/cwlog_client.go

```diff
@@ -34,6 +38,17 @@ type Client struct {
 	tags         map[string]*string
 	logger       *zap.Logger
 }
+type UserAgentOption func(*UserAgentFlag)
+
+type UserAgentFlag struct {
+	isAppSignals bool
+}
+
+func WithEnabledAppSignals(flag bool) UserAgentOption {
+	return func(ua *UserAgentFlag) {
+		ua.isAppSignals = flag
+	}
+}
```

Comment:
Could you make this more generalizable? Rather than only control adding a string for AppSignals it could just add any string passed into it to the user agent. The client doesn't need to know about AppSignals or not, just that it needs to add an extra string to the user agent.

Reply:
We may be left with a bit of tech debt if we do this though with the container insights logic in the user agent builder. I think that's okay™ for now though. Realistically a user could provide any number of extra strings to append to the user agent.

Reply:
Sounds good, no need in previous logic to have a "WithOption" for every single additional agent string.
With your suggestion, I replaced this with: `WithUserAgentExtras(userAgentExtras ...string`)